### PR TITLE
Only catch user errors for user params

### PR DESCRIPTION
**Pull request by drapergeek**

### CODE REVIEW

File: test/unit/cog_api/fake/helpers_test.exs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+defmodule CogApi.Fake.HelpersTest do
+  use CogApi.FakeCase
+
+  import CogApi.Fake.Helpers
+
+  describe "catch_errors" do
+    context "when given a struct as the first argument" do
+      it "will cause an error for any field with a value of ERRROR" do
```
Comment (jsteiner) [inline test/unit/cog_api/fake/helpers_test.exs]: `ERROR`

**Comment by jsteiner** [comment]
This seems like something we want for all fields. Should we remove the old `catch_errors` and add the struct to everywhere we use it? (probably only ~4-5 places right now)

**Comment by drapergeek** [comment]
I considered that and talked myself out of it but if you're on board, I'll do that.

**Comment by drapergeek** [comment]
@jsteiner any more thoughts?

**Comment by jsteiner** [comment]
👍

**Pull request merged**